src/embedded/index.ts

```diff
@@ -0,0 +1,301 @@
+/* eslint-disable no-sync */
+import fs from 'fs';
+import { get } from 'https';
+import net from 'net';
+import { spawn } from 'child_process';
+import { dirname } from 'path/posix';
+import { homedir } from 'os';
+import { join } from 'path';
+import { extract } from 'tar';
+
+const defaultBinaryPath = join(homedir(), '.cache/weaviate-embedded');
+const defaultPersistenceDataPath = join(homedir(), '.local/share/weaviate');
+const defaultVersion = '1.18.0';
+
+interface EmbeddedOptionsConfig {
+  host?: string;
+  port?: number;
+  env?: object;
+  version?: string;
+}
+
+export class EmbeddedOptions {
+  binaryPath: string;
+  persistenceDataPath: string;
+  host: string;
+  port: number;
+  clusterHostname: string;
+  version: string;
+  env: NodeJS.ProcessEnv;
+
+  constructor(cfg?: EmbeddedOptionsConfig) {
+    this.clusterHostname = 'embedded';
+    this.host = (cfg && cfg.host) || '127.0.0.1';
+    this.port = (cfg && cfg.port) || 6666;
```

Comment:
should be here some random port? or we agreed to make it default to `6666`?

Reply:
Unfortunately getting a random available port in nodes requires an async function, which would mean creating an `EmbeddedOptions` instance needs to be `await`'ed.

This is the same default port which the python client embedded db uses.

Reply:
ok, let's leave it then